release/nightly_tests/dataset/sort.py

```diff
@@ -0,0 +1,162 @@
+                ),
+                names=[f"c_{i}" for i in range(num_columns)],
+            )
+
+        schema = pyarrow.Table.from_pydict(
+            {f"c_{i}": [0] for i in range(num_columns)}
+        ).schema
+
+        i = 0
+        while i < n:
+            count = min(block_size, n - i)
+            meta = BlockMetadata(
+                num_rows=count,
+                size_bytes=8 * count * num_columns,
+                schema=schema,
+                input_files=None,
+                exec_stats=None,
+            )
+            read_tasks.append(
+                ReadTask(
+                    lambda count=count, num_columns=num_columns: [
+                        make_block(count, num_columns)
+                    ],
+                    meta,
+                )
+            )
+            i += block_size
+
+        return read_tasks
+
+
+if __name__ == '__main__':
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--num-partitions", help="number of partitions", default="50", type=str
+    )
+    parser.add_argument(
+        "--partition-size",
+        help="partition size (bytes)",
+        default="200e6",
+        type=str,
+    )
+    parser.add_argument(
+        "--shuffle", help="shuffle instead of sort", action="store_true"
+    )
+
+    args = parser.parse_args()
+
+    num_partitions = int(args.num_partitions)
+    partition_size = int(float(args.partition_size))
+    print(f"Dataset size: {num_partitions} partitions, {partition_size / 1e9}GB partition size, {num_partitions * partition_size / 1e9}GB total")
+    start_time = time.time()
+    source = RandomIntRowDatasource()
+    num_rows_per_partition = partition_size // 8
+    ds = ray.data.read_datasource(source,
+            parallelism=num_partitions,
+            n=num_rows_per_partition * num_partitions,
+            num_columns=1)
+    exc = None
+    try:
+        if args.shuffle:
+            ds = ds.random_shuffle()
+        else:
+            ds = ds.sort(key="c_0")
+    except Exception as e:
+        exc = e
+        pass
+
+    end_time = time.time()
+
+    print("Finished in", end_time - start_time)
+    print("")
+
+    print("==== Driver memory summary ====")
+    maxrss = int(resource.getrusage(resource.RUSAGE_SELF).ru_maxrss * 1e3)
+    print(f"max: {maxrss / 1e9}/GB")
+    process = psutil.Process(os.getpid())
+    rss = int(process.memory_info().rss)
+    print(f"rss: {rss / 1e9}/GB")
+
+    print(memory_summary(stats_only=True))
+    print("")
+
+    print(ds.stats())
+
+    if "TEST_OUTPUT_JSON" in os.environ:
+        out_file = open(os.environ["TEST_OUTPUT_JSON"], "w")
+        results = {
+                "time": end_time - start_time,
+                "success": "1" if exc is None else "0",
+                "num_partitions": num_partitions,
+                "partition_size": partition_size,
+                "perf_metrics": [
```

Comment:
@jjyao I'll try running it, but can you also look at this and make sure I'm reporting the metrics correctly for the dashboard?

Reply:
Should we also have a perf metric for runtime?

Reply:
Ah yes, I didn't realize the total run time wouldn't be enough. What do you put for the metric type in that case?

Reply:
"LATENCY"